### PR TITLE
Latest Course Finder

### PR DESCRIPTION
**Pull request by jalejandro3**
PR con caso de Uso LastCourseFinder del curso Arquitectura Hexagonal.
No tengo mucha experiencia haciendo pruebas, para esta feature lo que hubiese tenido mas sentido es crear una collection de Courses y buscar la última de esa collection.

### CODE REVIEW

File: src/Mooc/Courses/Application/Find/LastCourseFinder.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        $this->repository = $repository;
+    }
+
+    public function __invoke(): Course
+    {
+        $latestCourse = $this->repository->searchLatest();
+
+        if (is_null($latestCourse)) {
+            throw new CourseEmpty();
+        }
```
Comment (rgomezcasas) [inline src/Mooc/Courses/Application/Find/LastCourseFinder.php:26]: Extract to a method

File: src/Mooc/Courses/Domain/Course.php [hunk 2]
```diff
@@ -12,12 +12,14 @@ final class Course extends AggregateRoot
     private CourseId       $id;
     private CourseName     $name;
     private CourseDuration $duration;
+    private CourseCreatedAt $createdAt;
```
Comment (rgomezcasas) [inline src/Mooc/Courses/Domain/Course.php:15]: align

File: src/Mooc/Courses/Domain/Course.php [hunk 3]
```diff
@@ -12,12 +12,14 @@ final class Course extends AggregateRoot
...
 
     public function __construct(CourseId $id, CourseName $name, CourseDuration $duration)
     {
         $this->id       = $id;
         $this->name     = $name;
         $this->duration = $duration;
+        $this->createdAt = new CourseCreatedAt(\DateTimeImmutable::createFromMutable(new \DateTime('now')));
```
Comment (rgomezcasas) [inline src/Mooc/Courses/Domain/Course.php:22]: align

Comment (rgomezcasas) [reply src/Mooc/Courses/Domain/Course.php:22]: Also, move `new \DateTime('now')` to the `create`  method. If it's here, we won't be able to recreate it correctly on tests.
```php
public function __construct(CourseId $id, CourseName $name, CourseDuration $duration, DateTimeImmutable $createdAt)
```

File: tests/src/Mooc/Courses/Application/Find/LastCourseFinderTest.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    /** @test */
+    public function it_should_find_latest_course(): void
+    {
+        $latestCourse = CourseMother::random();
+
+        $this->shouldSearchLatest($latestCourse);
+
+        $this->finder->__invoke();
```
Comment (rgomezcasas) [inline tests/src/Mooc/Courses/Application/Find/LastCourseFinderTest.php:29]: With the change in the constructor, you'll be able to easily test it

**Review by rgomezcasas — CHANGES_REQUESTED** [review]
👏👏👏

Now try with a different approach, try to create a `last_course` module that changes every `course created` is published!

Thanks!!

File: src/Mooc/Courses/Application/Find/LastCourseFinder.php (hunk 1 above)
Comment (EnzoDosSantos) [reply src/Mooc/Courses/Application/Find/LastCourseFinder.php:26]: Hi, i would like to know why this need a method, ty.

**Pull request closed without merging**